helpers/calendar/analytics_helper.py

```diff
@@ -175,16 +176,37 @@ def get_room_statistics(self, number_of_events_in_room, all_details):
         return result
 
     @staticmethod
-    def get_total_bookings(self, query, start_date, end_date):
+    def get_total_bookings(instance, query, start_date, end_date, room_id=None):
         bookings = 0
-        rooms = CommonAnalytics.get_room_details(self, query)
+        rooms = CommonAnalytics.get_room_details(instance, query)
+        active_rooms = RoomModel.query.filter(RoomModel.state == "active")
         for room in rooms:
-            all_events = CommonAnalytics.get_all_events_in_a_room(
-                self, room["room_id"], start_date, end_date)
-            if all_events:
-                bookings += len(all_events)
+            if room_id:
+                exact_room = active_rooms.filter(
+                    RoomModel.id == room_id).first()
+                if not exact_room:
+                    raise GraphQLError("Room Id does not exist")
+                if room.get('calendar_id') == exact_room.calendar_id:
+                    bookings = CommonAnalytics.get_bookings_count(instance,
```

Comment:
I believe this section can be formatted better, those are a lot of unnecessary lines wasted. What do you think?

Reply:
Well, I think these lines are necessary because first we have to cater for a scenario where the room Id passed was wrong, then cater for a scenario where a correct Id was passed and hence the results required pertain to a single room. After this, we then proceed to cater for a scenario where no id was passed and hence results returned pertain to all rooms. How do you think it could be made better while maintaining this functionality? @verenceLola

Reply:
something like this, 
```
bookings = CommonAnalytics.get_bookings_count(
               instance, room, start_date, end_date, bookings
           )
```

Reply:
thanx @verenceLola though with this, there is no way to confirm that a particular `calendar_id` actually exists. this would work if this function was only returning bookings for all rooms in a particular location which already does though it was simply modified to also take care for booking count for a single room in a location. lets us sync so I can try to clarify more.

Reply:
thanx @verenceLola this change has been implemented